langchain-core/src/prompts/chat.ts

```diff
@@ -307,6 +310,274 @@ export class ChatMessagePromptTemplate<
+  }
+
+  getRoleFromMessageClass(name: string) {
+    switch (name) {
+      case "HumanMessage":
+        return "human";
+      case "AIMessage":
+        return "ai";
+      case "SystemMessage":
+        return "system";
+      case "ChatMessage":
+        return "chat";
+      default:
+        throw new Error("Invalid message class name");
+    }
+  }
+
+  static fromTemplate(
+    template: string | Array<string | _TextTemplateParam | _ImageTemplateParam>,
+    additionalOptions?: Record<string, unknown>
+  ) {
+    if (typeof template === "string") {
+      return new this(PromptTemplate.fromTemplate(template));
+    }
+    const prompt: Array<
+      PromptTemplate<InputValues> | ImagePromptTemplate<InputValues>
+    > = [];
+    for (const item of template) {
+      if (
+        typeof item === "string" ||
+        (typeof item === "object" && "text" in item)
+      ) {
+        let text = "";
+        if (typeof item === "string") {
+          text = item;
+        } else if (typeof item.text === "string") {
+          text = item.text ?? "";
+        } else {
+          // handle
+        }
+        prompt.push(PromptTemplate.fromTemplate(text));
+      } else if (typeof item === "object" && "image_url" in item) {
+        let imgTemplate = item.image_url ?? "";
+        let imgTemplateObject: ImagePromptTemplate<InputValues>;
+        let inputVariables: string[] = [];
+        if (typeof imgTemplate === "string") {
+          const parsedTemplate = parseFString(imgTemplate);
+          const variables = parsedTemplate.flatMap((item) =>
+            item.type === "variable" ? [item.name] : []
+          );
+
+          if (variables) {
+            if (variables.length > 1) {
+              throw new Error(
+                `Only one format variable allowed per image template.\nGot: ${variables}\nFrom: ${imgTemplate}`
+              );
+            }
+            inputVariables = [variables[0]];
+          } else {
+            inputVariables = [];
+          }
+
+          imgTemplate = { url: imgTemplate };
+          imgTemplateObject = new ImagePromptTemplate<InputValues>({
+            template: imgTemplate,
+            inputVariables
+          });
+        } else if (typeof imgTemplate === "object") {
+          if ("url" in imgTemplate) {
+            const parsedTemplate = parseFString(imgTemplate.url);
+            inputVariables = parsedTemplate.flatMap((item) =>
+              item.type === "variable" ? [item.name] : []
+            );
+          } else {
+            inputVariables = [];
+          }
+          imgTemplateObject = new ImagePromptTemplate<InputValues>({
+            template: imgTemplate,
+            inputVariables
+          });
+        } else {
+          throw new Error("Invalid image template");
+        }
+        prompt.push(imgTemplateObject);
+      }
+    }
+    // @ts-expect-error todo fix
+    return new this({ prompt, additionalOptions });
```

Comment:
The logic for the typing here is very tricky, we haven't figured out anything good for `ChatPromptTemplate.fromMessages()` yet which this looks similar to.

I would just use `any` for now.